pip/resolve.py

```diff
@@ -296,12 +308,14 @@ def _resolve_one(self, requirement_set, req_to_install,
                 else:
                     logger.info('Collecting %s', req_to_install)
 
+        assert self.require_hashes is not None
```

Comment:
It'd be nice to either have a comment or using the ``assert <expression>, "An Error Message"`` to mention what invariant would trigger this to fail.